timemachine/cpp/src/barostat.cu

```diff
@@ -134,9 +136,9 @@ void __global__ rescale_positions(
 
     const double num_atoms = static_cast<double>(mol_offsets[mol_idx+1] - mol_offsets[mol_idx]);
 
-    const double centroid_x = centroids[mol_idx*3+0] / num_atoms;
-    const double centroid_y = centroids[mol_idx*3+1] / num_atoms;
-    const double centroid_z = centroids[mol_idx*3+2] / num_atoms;
+    const double centroid_x = FIXED_TO_FLOAT<double>(centroids[mol_idx*3+0]) / num_atoms;
```

Comment:
you probably don't want to use this particular FIXED_TO_FLOAT, it has a precision designed for the nonbonded forces.

Reply:
Which fixed precision function is preferable? Do we want another level of fixed precision for positions?